public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/useAdvancedMode.ts

```diff
@@ -47,14 +46,15 @@ export function determineAdvancedMode(
   depending on the editor settings, the alert type, and the queries.
    */
 export const useAdvancedMode = (
-  editorSettings: SimplifiedEditor | undefined,
+  simplifiedQueryEditor: boolean | undefined,
   isGrafanaAlertingType: boolean,
   isNewFromQueryParams: boolean,
   dataQueries: Array<AlertQuery<ExpressionQuery | AlertDataQuery>>,
   expressionQueries: Array<AlertQuery<ExpressionQuery>>
 ) => {
+  const { setValue } = useFormContext<RuleFormValues>();
```

Comment:
I see some inconsistency here. We pass `simplifiedQueryEditor` as a parameter (which is effectively `editorSettings.simplifiedQueryEditor`) but at the same time we add a formContext as a dependency
If we want to keep the form context as a dependency, we don't need to pass most of the params. On the other hand I'd try not to have form dependencies in hooks

Reply:
this code has been simplified